pkg/controller/multi_writer_at.go

```diff
@@ -33,30 +33,37 @@ func (m *MultiWriterError) Error() string {
 	}
 }
 
-func (m *MultiWriterAt) WriteAt(p []byte, off int64) (n int, err error) {
+func (m *MultiWriterAt) WriteAt(p []byte, off int64) (int, error) {
 	errs := make([]error, len(m.writers))
-	errored := false
 	wg := sync.WaitGroup{}
 
 	for i, w := range m.writers {
 		wg.Add(1)
 		go func(index int, w io.WriterAt) {
 			_, err := w.WriteAt(p, off)
 			if err != nil {
-				errored = true
 				errs[index] = err
 			}
+
 			wg.Done()
 		}(i, w)
 	}
 
 	wg.Wait()
-	if errored {
-		return 0, &MultiWriterError{
-			Writers: m.writers,
-			Errors:  errs,
+
+	n := 0
+	var err error = nil
+
+	for i := range errs {
+		if errs[i] != nil {
+			err = &MultiWriterError{
+				Writers: m.writers,
+				Errors:  errs,
+			}
```

Comment:
This function can `break` from this loop if there is an error. 
BTW, any reason for removing the boolean `errored` and using this loop instead?

Reply:
Make sense. Thank you for your explanation.